crates/bevy_asset/src/io/android.rs

```diff
@@ -56,18 +55,47 @@ impl AssetReader for AndroidAssetReader {
 
     fn read_directory<'a>(
         &'a self,
-        _path: &'a Path,
+        path: &'a Path,
     ) -> BoxedFuture<'a, Result<Box<PathStream>, AssetReaderError>> {
-        let stream: Box<PathStream> = Box::new(EmptyPathStream);
-        error!("Reading directories is not supported with the AndroidAssetReader");
-        Box::pin(async move { Ok(stream) })
+        Box::pin(async move {
+            let asset_manager = bevy_winit::ANDROID_APP
+                .get()
+                .expect("Bevy must be setup with the #[bevy_main] macro on Android")
+                .asset_manager();
+            let opened_assets_dir = asset_manager
+                .open_dir(&CString::new(path.to_str().unwrap()).unwrap())
```

Comment:
I don't know to be honest. Not sure if that is widely supported in Android. Either way, `to_str().unwrap()` is used for all functions for `AndroidAssetReader`, so maybe it would be good to investigate that separately and fix if needed.

Reply:
I tried to check if there is any assumption made regarding app asset paths on Android but couldn't find anything. In any case you're right that this way is consistent with the already implemented functions, so that can be adjusted later if need be.

Reply:
I was just going by the way most apps are written (Java / Kotlin) and they assume all paths are valid UTF-8, unless I misunderstood this: https://developer.android.com/reference/java/io/File#interoperability-with-java.nio.file-package